ocs_ci/utility/metadata_utils.py

```diff
@@ -0,0 +1,308 @@
+"""
+Module that contains all operations related to add metadata feature in a cluster
+
+"""
+
+import logging
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources import pod
+from ocs_ci.utility.retry import retry
+from ocs_ci.ocs.exceptions import (
+    CommandFailed,
+    ResourceWrongStatusException,
+)
+
+log = logging.getLogger(__name__)
+
+
+def check_setmetadata_availability(pod_obj):
+    """
+    Check setmetadata parameter is available or not for cephfs and rbd plugin pods
+
+    """
+    plugin_provisioner_pod_objs = pod.get_all_pods(
+        namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+        selector=["csi-cephfsplugin-provisioner", "csi-rbdplugin-provisioner"],
+    )
+    log.info(f"list of provisioner pods---- {plugin_provisioner_pod_objs}")
+    response = retry((CommandFailed, ResourceWrongStatusException), tries=3, delay=15)(
+        pod.validate_pods_are_respinned_and_running_state
+    )(plugin_provisioner_pod_objs)
+    log.info(response)
+    get_args_provisioner_plugin_pods = []
+    for plugin_provisioner_pod in plugin_provisioner_pod_objs:
+        containers = pod_obj.exec_oc_cmd(
+            "get pod "
+            + plugin_provisioner_pod.name
+            + " --output jsonpath='{.spec.containers}'"
+        )
+        for entry in containers:
+            if "--setmetadata=true" in entry["args"]:
+                get_args_provisioner_plugin_pods.append(entry["args"])
+    if get_args_provisioner_plugin_pods:
+        return all(
+            ["--setmetadata=true" in args for args in get_args_provisioner_plugin_pods]
+        )
+    else:
+        return False
+
+
+def enable_metadata(
+    config_map_obj,
+    pod_obj,
+):
+    """
+    Enable CSI_ENABLE_METADATA
+    Steps:
+    1:- Enable CSI_ENABLE_METADATA flag via patch request
+    2:- Check csi-cephfsplugin provisioner and csi-rbdplugin-provisioner
+    pods are up and running
+    3:- Check 'setmatadata' is set for csi-cephfsplugin-provisioner
+    and csi-rbdplugin-provisioner pods
+
+    Returns:
+        str: cluster name
+
+    """
+    enable_metadata = '{"data":{"CSI_ENABLE_METADATA": "true"}}'
+
+    # Enable metadata feature for rook-ceph-operator-config using patch command
+    assert config_map_obj.patch(
+        resource_name="rook-ceph-operator-config",
+        params=enable_metadata,
+    ), "configmap/rook-ceph-operator-config not patched"
+
+    # Check csi-cephfsplugin provisioner and csi-rbdplugin-provisioner pods are up and running
+    assert pod_obj.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector="app=csi-cephfsplugin-provisioner",
+        dont_allow_other_resources=True,
+        timeout=60,
+    )
```

Comment:
assert massage is missing. Please add to other places where it's missing

Reply:
Addressed in latest commit.